STUD-178/profile, CODE SMELL reduced from 44 to 10

adamyafout opened the pull request:
some code smells still exist need to discuss them with team lead

adamyafout committed:
STUD-178/profile, CODE SMELL reduced from 44 to 10

some code smells still exist need to discuss them with team lead

adamyafout committed:
STUD-178/profile, CODE SMELL reduced from 44 to 10

some code smells still exist need to discuss them with team lead

car12lin12 commented on Review/delivery/http/review_delivery_test.go:
```diff
@@ -19,7 +19,13 @@ import (
 )
 
-func TestReviewHandler_AddReview(t *testing.T) {
+const stardomainreview = "*domain.Review"
```
please use camel case for these private constant variables to follow the golang naming convention, and maybe add the term "path" to the strings that contain paths. Here are my suggestions for the fixes:
sapireviews -> postReviewPath
apireviewsupdate -> putReviewPath
stardomainreview -> reviewType
failuremessage -> failureMessage
applicationJSON is okay

car12lin12 commented on Review/usecase/review_usecase_test.go:
```diff
@@ -14,6 +14,8 @@ import (
 	"github.com/stretchr/testify/mock"
 )
 
+const casesuccess = "case success"
```
camel case naming convention:
casesuccess -> caseSuccess
stardomainreview -> reviewType

car12lin12 commented on School/delivery/http/school_delivery_test.go:
```diff
@@ -11,16 +11,20 @@ import (
 )
 
+const failureMessage = "failed to read from message"
+const sapischoolconfirmemail = "%s/api/school/confirm?email=%s"
```
follow camel case convention:
sapischoolconfirmemail -> postSchoolEmailConfirmationPath
testemail -> testEmail

car12lin12 commented on Student/delivery/http/student_http_test.go:
```diff
@@ -19,7 +19,14 @@ import (
-func TestStudentHandler_GetByID(t *testing.T) {
+
+const failureMessage = "failed to read from message"
+const domaindotstudent = "*domain.Student"
```
camel case naming convention and naming suggestions:
domaindotstudent -> studentType
apistudents -> putStudentPath
sapistudent -> postStudentPath
sapistudents -> getStudentPath

car12lin12 commented on Student/usecase/student_usecase_test.go:
```diff
@@ -20,13 +20,14 @@ func TestCreate(t *testing.T) {
 	var mockStudent domain.Student
 	faker.FakeData(&mockStudent)
 
+	const domainstudent = "*domain.Student"
```
camel case naming convention & naming suggestion:
domainstudent -> studentType

car12lin12 commented on app/urls.go:
```diff
@@ -11,10 +11,11 @@ import (
 	authorized := router.Group("/api")
 	authorized.Use(m.AuthMiddleware())
-	authorized.GET("/student/:id", h.GetByID)
+	const studentidpath = "/student/:id"
```
camel case naming convention:
studentidpath -> pathStudentID

car12lin12 reviewed (changes requested):
Nice job Adam!

My comments concern following the camel case naming convention and a few naming suggestions!

adamyafout committed:
STUD-178/profile, using camel case convention

the variable naming convention

adamyafout committed:
STUD-178/profile, using camel case convention

the variable naming convention

car12lin12 commented on Student/usecase/student_usecase.go:
```diff
@@ -35,7 +38,7 @@ func (s *studentUseCase) Create(c context.Context, st *domain.Student) error {
 		if err == nil {
 			if !reflect.DeepEqual(*existingStudent, domain.Student{}) {
-				return errors.NewConflictError(fmt.Sprintf("Student with ID %s already exists", st.ID))
+				return errors.NewConflictError(fmt.Sprintf(errorMessage, st.ID))
```
this should be a different error message indicating the student already exists instead of does not exist

nuzurie committed:
STUD-134, STUD-207, STUD-208: bug fixes

return student body on update, fix order of CORS to allow cross-origin access, ensure review body when posting has at least one tag

nuzurie committed:
fix tests

adamyafout committed:
STUD-178/profile, Fixing the wrong error message

adamyafout committed:
STUD-178/profile, Fixing the wrong error message

adamyafout committed:
Merge remote-tracking branch 'origin/refactor/resolve-technical-debt' into refactor/resolve-technical-debt
